RCNN_tf.py

```diff
@@ -396,93 +548,168 @@ def train(
 
                     total_parameters += variable_parameters
 
-                print "Total number of trainable parameters:", total_parameters, "\n"
+                print "\nTotal number of trainable parameters:", total_parameters, "\n"
 
-            total_batch = int(training_data_features.shape[0] / batch_size)
+            total_batch = int(train_data_feats.shape[0] / batch_size)
             for epoch in range(epochs):
                 avg_cost = 0
                 sess.run(self.train_init_op, feed_dict={
-                    self.input_placeholder: training_data_features,
-                    self.output_placeholder: training_data_labels,
+                    self.input_placeholder: train_data_feats,
+                    self.output_placeholder: train_data_labels,
                     self.num_data_placeholder: batch_size
                 })
                 for i in range(total_batch):
                     progress = (i / float(total_batch - 1)) * 100
                     print '\r {:.1f}%'.format(progress), '\t{0}> '.format('#' * int(progress)),
 
-                    accuracies, _, cost_ = sess.run([self.summaries, self.optimiser, self.cross_entropy],
-                                                    feed_dict={
-                                                        self.rate_placeholder: 0.2,
-                                                        self.num_data_placeholder: batch_size,
-                                                    })
+                    accuracies, _, cost_ = sess.run(
+                        [self.summaries, self.optimiser, self.cross_entropy],
+                        feed_dict={
+                            self.rate_placeholder: 0.2,
+                            self.num_data_placeholder: batch_size,
+                        }
+                    )
+
                     avg_cost += cost_ / total_batch
                     train_writer.add_summary(accuracies)
 
                 sess.run(self.test_init_op, feed_dict={
-                    self.input_placeholder: val_data_features,
+                    self.input_placeholder: val_data_feats,
                     self.output_placeholder: val_data_labels,
-                    self.num_data_placeholder: val_data_features.shape[0]
+                    self.num_data_placeholder: val_data_feats.shape[0]
```

Comment:
Again: Fine for now, but this can be very dangerous and break things.

Reply:
I did a thorough test, so don't worry.